trackintel/preprocessing/trips.py

```diff
@@ -0,0 +1,322 @@
+                new_list_start = j + 1
+                continue
+
+            # check whether the start-end candidate of a tour is an unkown activity
+            if pd.isna(user_trip_df.loc[cand, "origin_staypoint_id"]):
+                continue
+
+            # TODO: compute length of triplegs and sum - must be larger than minthresh
+            # check if endpoint of trip = start location of cand
+            if stps_w_locs is not None:
+                end_start_at_same_loc = _check_same_loc(
+                    user_trip_df.loc[cand, "origin_staypoint_id"],  # start stp of first trip
+                    row["destination_staypoint_id"],  # destination stp of current trip
+                    stps_w_locs,
+                )
+            else:
+                # check distance between point 1: end point of current trip, point 2: start of first trip on tour
+                end_start_at_same_loc = _check_max_dist(row["geom"][1], user_trip_df.loc[cand, "geom"][0], max_dist)
+
+            # print("Check distance to start", cand, end_point, cand_start_point, point_dist)
+            if end_start_at_same_loc:
+                # Tour found!
+                # collect the trips on the tour in a list
+                non_gap_trip_idxs = [c for c in start_candidates[j:] if ~np.isnan(c)]
+                tour_candidate = user_trip_df[user_trip_df.index.isin(non_gap_trip_idxs)]
+                tours.append(_create_tour_from_stack(tour_candidate, stps_w_locs, max_dist, max_time))
+
+                nr_gaps = np.sum(np.isnan(np.array(start_candidates[j:])))
+
+                if nr_gaps > max_gap_size:
+                    # No tour found, too many gaps inbetween
+                    continue
+                # print("Tour found!", tour_candidate.head())
+
+                # remove trips that were on the tour
+                start_candidates = start_candidates[:j]
+                # remove gap if there is a gap in the end
+                if np.isnan(start_candidates[-1]):
+                    del start_candidates[-1]
+                # do not consider the other trips - one trip cannot close two tours at a time anyway
+                break
+
+        # remove points because they are out of the time window
+        start_candidates = start_candidates[new_list_start:]
+
+    tours_df = pd.DataFrame(tours)
+    return tours_df
+
+
+def _check_same_loc(stp1, stp2, stps_w_locs):
+    if pd.isna(stp1) or pd.isna(stp2):
+        return False
+    return stps_w_locs.loc[stp1, "location_id"] == stps_w_locs.loc[stp2, "location_id"]
+
+
+def _check_max_dist(p1, p2, max_dist):
+    """
+    Check whether two points p1, p2 are less or equal than max_dist apart
+
+    Parameters
+    --------
+    p1, p2: shapely Point objects
+    max_dist: int
+
+    Returns
+    ------
+    dist_below_thresh: bool
+        indicating whether p1 and p2 are less than max_dist apart
+    """
+    geod = Geod(ellps="CPM")  # other guy used WGS84
```

Comment:
Ah, I don't know if "CPM" is a good choice as ellipse (I actually don't know ;-) ) but distance calculation should be coordinate system aware. Even though we use WGS84 most of the time we also want to support projected (=local) coordinate systems. It is OK to use the pyproj functions for WGS84 coordinate systems but for planar coordinate systems it is easier to use the euclidean distance [provided by shapely](https://shapely.readthedocs.io/en/stable/manual.html#object.distance). Best would be if you use our helper function [check_gdf_crs] for that but I just noticed an [issue](https://github.com/mie-lab/trackintel/issues/288) with this function. We also have a vectorized version of the [haversine distance ](https://github.cohttps://github.com/mie-lab/trackintel/blob/563473b885a91858a8463da9db66353d5adb1089/trackintel/geogr/distances.py#L186/mie-lab/trackintel/blob/master/trackintel/geogr/point_distances.py).
We don't need to support non-planar coordinate systems other than WGS84.

Reply:
Okay I think I should read a bit about coordinate systems, because I don't know much about it tbh :D I changed it to use the trackintel `haversine_dist` function now, is that okay?